guide/samples/tests/mocking_lros_error.rs

```diff
@@ -0,0 +1,217 @@
+        Ok(operation)
+    }
+    // ANCHOR_END: error-op
+
+    #[tokio::test]
+    async fn error_starting_lro() -> Result<()> {
+        let mut mock = MockSpeech::new();
+        // ANCHOR: expectation-initial
+        mock.expect_batch_recognize()
+            .return_once(|_, _| Err(Error::other("failed to start operation")));
+        // ANCHOR_END: expectation-initial
+
+        // Create a client, implemented by our mock.
+        let client = speech::client::Speech::from_stub(mock);
+
+        // Call our function which manually polls.
+        let result = my_manual_poller(&client, "my-project").await;
+
+        // Verify the the final result.
+        assert!(result.billed_duration.is_err());
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn lro_ending_in_error() -> Result<()> {
+        let mut seq = mockall::Sequence::new();
+        let mut mock = MockSpeech::new();
+        mock.expect_batch_recognize()
+            .once()
+            .in_sequence(&mut seq)
+            .returning(|_, _| make_partial_operation(25));
+        mock.expect_get_operation()
+            .once()
+            .in_sequence(&mut seq)
+            .returning(|_, _| make_partial_operation(50));
+        mock.expect_get_operation()
+            .once()
+            .in_sequence(&mut seq)
+            .returning(|_, _| make_partial_operation(75));
+        // ANCHOR: expectation-final
+        mock.expect_get_operation()
+            .once()
+            .in_sequence(&mut seq)
+            .returning(|_, _| {
+                // This is a common error for `Create*` RPCs, which are often
+                // LROs. It is less applicable to `BatchRecognize` in practice.
+                let status = rpc::model::Status::default()
+                    .set_code(gax::error::rpc::Code::AlreadyExists as i32)
+                    .set_message("resource already exists");
+                make_failed_operation(status)
+            });
+        // ANCHOR_END: expectation-final
+
+        // Create a client, implemented by our mock.
+        let client = speech::client::Speech::from_stub(mock);
+
+        // Call our function which manually polls.
+        let result = my_manual_poller(&client, "my-project").await;
+
+        // Verify the partial metadata updates, and the final result.
+        assert_eq!(result.progress_updates, [25, 50, 75]);
+        assert!(result.billed_duration.is_err());
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn polling_loop_error() -> Result<()> {
+        let mut seq = mockall::Sequence::new();
+        let mut mock = MockSpeech::new();
+        mock.expect_batch_recognize()
+            .once()
+            .in_sequence(&mut seq)
+            .returning(|_, _| make_partial_operation(25));
+        // ANCHOR: expectation-polling-error
+        mock.expect_get_operation()
+            .once()
+            .in_sequence(&mut seq)
+            .returning(|_, _| Err(Error::other("could not poll operation")));
```

Comment:
I think it would be more interesting to return realistic errors here, like a `ServiceError` with `Code::Aborted` or something.  If you are writing tests that return errors, it is because your code handles realistic errors, we need to feed it realistic errors too. If not, we should explain that we are using `Error::other` to simplify the tests, and that real RPCs return something different.

Reply:
Done.